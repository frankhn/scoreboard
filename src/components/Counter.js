import PropTypes from "prop-types";
import React from "react";

const Counter = ({ index, score, changerScore }) => {
  return (
    <div className="counter">
      <button
        id="decrement"
        className="counter-action decrement"
        onClick={() => changerScore(index, -1)}
      >
        -
      </button>
      <span className="counter-score">{score}</span>
      <button
        className="counter-action increment"
        onClick={() => changerScore(index, 1)}
      >
        +
      </button>
    </div>
  );
};

Counter.propTypes = {
  index: PropTypes.number,
  score: PropTypes.number,
  changerScore: PropTypes.func
};

export default Counter;
