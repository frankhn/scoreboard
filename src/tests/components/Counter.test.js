import "../../../enzyme.config";
import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import Counter from "../../components/Counter";

const changerScore = jest.fn();
const props = {
  index: 0,
  score: 2,
  changerScore
};
const component = shallow(<Counter {...props} />);
const instance = component.instance();

afterEach(() => {
  changerScore.mockClear();
});
describe("Render counter", () => {
  test("should render counter component to match the snapshot", () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  test("should call changerScore function", () => {
    component
      .find(`[id="decrement"]`)
      .at(0)
      .simulate("click");
    expect(changerScore).toHaveBeenCalled();
    console.log("====== =======", component.props);
    // expect(component.props.changerScore).toHaveBeenCalled();
  });
});
