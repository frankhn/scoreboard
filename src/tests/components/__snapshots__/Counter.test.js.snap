// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render counter should render counter component to match the snapshot 1`] = `
<div
  className="counter"
>
  <button
    className="counter-action decrement"
    id="decrement"
    onClick={[Function]}
  >
    -
  </button>
  <span
    className="counter-score"
  >
    2
  </span>
  <button
    className="counter-action increment"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
